DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataStructures.Lists 
+{
+    public class CircularBuffer<T> : IEnumerable<T> , ICollection<T> where T : IComparable<T> {
+        private T[] _circularBuffer;
+        private int _end;
+        private int _start;
+        private int _length { get => _circularBuffer.Length - 1; }
+        private readonly T _default;
+        private static readonly int _defaultBufferLength = 10;
+
+        /// <summary>
+        /// Controls whether data should be overridden when it is continously inserted without reading
+        /// </summary>
+        public bool CanOverride { get; }
+        public bool IsEmpty { get => _end == _start; }
```

Comment:
Does this make sense if it's a circular buffer with a `CanOverride=true` case? What would `IsEmpty` mean here?

Reply:
Please review the new Implementation. added a `_count` field that monitored each insertion, pop, and Removal of elements.